exonum/src/proto/schema/exonum/runtime.proto

```diff
@@ -157,8 +156,8 @@ message InstanceState {
   }
   // Service instance specification.
   exonum.runtime.InstanceSpec spec = 1;
-  // Service instance activity status.
+  // Service instance status.
   Status status = 2;
-  // Pending status of instance if the value is not `NONE`.
+  // Pending status of the instance.
```

Comment:
You removed assertion that the `Status::NONE` cannot be pending.

Reply:
🤔 Waaait. Does it mean that `pending_status` must never be equal to Status::NONE? If so, what is the purpose of Status::NONE?

Reply:
There is a comment `// Reserved for the further usage.` 🙂

Reply:
This status exists only between the transaction execution that initiates the instance status change and the block commit, thus it can only visible if you read dispatcher schema. But this status is still important to distinguish instance states during the block execution.
Purpose of Status::NONE is to avoid protobuf design flaws.